mapie/metrics.py

```diff
@@ -737,3 +740,578 @@ def hsic(
+    return ku_stat
+
+
+def kuiper_cdf(x: float) -> float:
+    """
+    Compute the Kuiper cumulative distribution function (CDF) for the float x.
+    This is interpreted as the CDF of the range
+    of the standard Brownian motion over the unit interval [0, 1].
+    The function is approximated by its power series, truncated so as to hit
+    machine precision error.
+
+    Parameters
+    ----------
+    x : float
+        The float x to compute the cumulative distribution function.
+
+    Returns
+    -------
+    float
+        The Kuiper cumulative distribution function.
+
+    References
+    ----------
+    Tygert M.
+    Calibration of P-values for calibration and for deviation
+    of a subpopulation from the full population.
+    arXiv preprint arXiv:2202.00100.
+    2022 Jan 31.
+
+    William Feller.
+    The Asymptotic Distribution of the Range of Sums of
+    Independent Random Variables.
+    Ann. Math. Statist. 22 (3) 427 - 432
+    September, 1951.
+
+    Example
+    -------
+    >>> import numpy as np
+    >>> from mapie.metrics import kuiper_cdf
+    >>> print(np.round(kuiper_cdf(1), 4))
+    0.0634
+    """
+    kmax = np.ceil(
+        (
+            0.5 + x / (np.pi * np.sqrt(2)) *
+            np.sqrt(
+                np.log(
+                    4 / (np.sqrt(2 * np.pi) * EPSILON) * (1 / x + x / np.pi**2)
+                )
+            )
+        )
+    )
+    c = 0.0
+    for k in range(int(kmax)):
+        kplus = k + 1 / 2
+        c += (
+            (8 / x**2 + 2 / kplus**2 / np.pi**2) *
+            np.exp(-2 * kplus**2 * np.pi**2 / x**2)
+        )
+    return c
+
+
+def kuiper_p_value(y_true: NDArray, y_score: NDArray) -> float:
+    """
+    Compute Kuiper statistic p-value.
+    Deduced from the corresponding statistic and CDF.
+    It represents the probability of the observed statistic
+    under the null hypothesis of perfect calibration.
+
+    Parameters
+    ----------
+    y_true : NDArray of shape (n_samples,)
+        An array of ground truth.
+
+    y_score : NDArray of shape (n_samples,)
+        An array of scores.
+
+    Returns
+    -------
+    float
+        The Kuiper p-value.
+
+    References
+    ----------
+    Tygert M.
+    Calibration of P-values for calibration and for deviation
+    of a subpopulation from the full population.
+    arXiv preprint arXiv:2202.00100.
+    2022 Jan 31.
+
+    William Feller.
+    The Asymptotic Distribution of the Range of Sums of
+    Independent Random Variables.
+    Ann. Math. Statist. 22 (3) 427 - 432
+    September, 1951.
+
+    Example
+    -------
+    >>> import pandas as pd
+    >>> from mapie.metrics import kuiper_p_value
+    >>> y_true = np.array([1, 0, 1, 0, 1, 0])
+    >>> y_score = np.array([0.8, 0.3, 0.5, 0.5, 0.7, 0.1])
+    >>> ku_p_value = kuiper_p_value(y_true, y_score)
+    >>> print(np.round(ku_p_value, 4))
+    0.9684
+    """
+    ku_stat = kuiper_statistic(y_true, y_score)
```

Comment:
You do not check for `column_or_1d`?

Reply:
The check is already done in `kuiper_statistic`. I wondered where to put the check : 
- in *_statistic functions
- or in *_p_value functions
I choose the first option, but both are valid, have no preference about it. What do you think ?